### DISCUSSION TITLE
Remove blank lines inside functions/methods

### DISCUSSION THREAD

**Discussion by natanfelles** [PHP-CS-Fixer/PHP-CS-Fixer#7306] [Q&A] [answered]
Hello! 

I would like to know if there is a rule to remove blank lines within functions and methods.

For example:

```php
class Foo
{
    public function makeFoo($foo)
    {
        echo 'foo'; // next line must be removed

        if ($foo) {
            return 1;
        } // next lines must be removed


        return 1;
    }
}
```

Thank you very much!

**Comment by Wirone** [comment]
No, we currently don't provide such ability. With `blank_line_before_statement` you can enforce blank line before configure statements, but if blank line is already there, then this fixer won't remove it even if you configure it with `"statements": []`. Why would you want to remove these empty lines?

**Reply by natanfelles** [reply]
I would like to decrease the height of the methods. Normally when a method gets too big, I create new methods. So I don't need blank lines inside the methods.

**Reply by natanfelles** [reply]
Ideas that made me stop using blank lines:

- https://phpmd.org/rules/codesize.html#excessivemethodlength
- https://www.yegor256.com/2014/11/03/empty-line-code-smell.html

**Reply by Wirone** [reply, marked as answer]
`ExcessiveMethodLength` can be configured with `ignore-whitespace=true` (this should be default IMHO). Also, I don't agree with Yegor's article, it's totally subjective - personally I am not comfortable with the condensed code, without empty lines. Anyway, we don't have such fixer currently and it's rather not possible that it will be available anytime soon.

**Reply by natanfelles** [reply]
OK. Thanks @Wirone !
